lib/kubeclient/common.rb

```diff
@@ -99,6 +99,10 @@ def initialize_client(
       end
     end
 
+    def with_faraday_config(&block)
```

Comment:
it looks like this could be called multiple times, but it can't ... so maybe call it `replace_faraday_config`
... and would seem more intuitive to specify this during initalization anyway like `configure_faraday: -> {|c| stuff }`